GPU/Common/ReplacedTexture.cpp

```diff
@@ -54,55 +54,68 @@ class ReplacedTextureTask : public Task {
 	LimitedWaitable *waitable_;
 };
 
+// This can only return true if ACTIVE or NOT_FOUND.
 bool ReplacedTexture::IsReady(double budget) {
 	_assert_(vfs_ != nullptr);
-	lastUsed_ = time_now_d();
-	if (threadWaitable_ && !threadWaitable_->WaitFor(budget)) {
-		return false;
-	}
 
-	// Loaded already, or not yet on a thread?
-	if (initDone_ && levelData_ && !levelData_->data.empty()) {
-		// TODO: lock?
-		levelData_->lastUsed = lastUsed_;
+	switch (State()) {
+	case ReplacementState::ACTIVE:
+	case ReplacementState::NOT_FOUND:
+		if (threadWaitable_) {
+			if (!threadWaitable_->WaitFor(budget)) {
+				return false;
+			}
+			// Successfully waited! Can get rid of it.
+			threadWaitable_->WaitAndRelease();
+			threadWaitable_ = nullptr;
+		}
+		lastUsed_ = time_now_d();
```

Comment:
This was previously set even if WaitFor failed, just to make sure it didn't expire with unfortunate timing (i.e. right after finishing load but before being read again.)

-[Unknown]

Reply:
Ah I see, yeah that makes sense. I'll restore it.